pkg/operator/staticresource/controller.go

```diff
@@ -0,0 +1,289 @@
+		"v4.1.0/aws-pod-identity-webhook/role.yaml",
+		"v4.1.0/aws-pod-identity-webhook/clusterrolebinding.yaml",
+		"v4.1.0/aws-pod-identity-webhook/rolebinding.yaml",
+		"v4.1.0/aws-pod-identity-webhook/svc.yaml",
+	}
+	templateFiles = []string{
+		"v4.1.0/aws-pod-identity-webhook/deployment.yaml",
+		"v4.1.0/aws-pod-identity-webhook/mutatingwebhook.yaml",
+	}
+	defaultCodecs = serializer.NewCodecFactory(scheme.Scheme)
+	defaultCodec  = defaultCodecs.UniversalDeserializer()
+)
+
+type staticResourceInitialApplyController struct {
+	reconciler *staticResourceReconciler
+}
+
+func (c *staticResourceInitialApplyController) Start(stopCh <-chan struct{}) error {
+	c.reconciler.ReconcileResources()
+	<-stopCh
+	return nil
+}
+
+func Add(mgr manager.Manager, kubeconfig string) error {
+	// Currently the only static assets are AWS only
+	infraStatus, err := platform.GetInfraStatusUsingKubeconfig(mgr, kubeconfig)
+	if err != nil {
+		return err
+	}
+	platformType := platform.GetType(infraStatus)
+	if platformType != configv1.AWSPlatformType {
+		return nil
+	}
+
+	log.Infof("Setting up static resource controller. Platform Type is %s", platformType)
+
+	config := mgr.GetConfig()
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	controllerRef := &corev1.ObjectReference{
+		Kind:      "deployment",
+		Namespace: operatorNamespace,
+		Name:      deploymentName,
+	}
+	eventRecorder := events.NewKubeRecorder(clientset.CoreV1().Events(operatorNamespace), deploymentName, controllerRef)
+
+	r := &staticResourceReconciler{
+		clientset:     clientset,
+		logger:        log.WithFields(log.Fields{"controller": controllerName}),
+		eventRecorder: eventRecorder,
+	}
+
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	p := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return isManaged(e.MetaNew)
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return isManaged(e.Meta)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return isManaged(e.Meta)
+		},
+	}
+
+	cache := mgr.GetCache()
+	allFiles := append(staticFiles, templateFiles...)
+	for _, file := range allFiles {
+		objBytes := v410_00_assets.MustAsset(file)
+		obj, _, err := defaultCodec.Decode(objBytes, nil, nil)
+		if err != nil {
+			return err
+		}
+
+		informer, err := cache.GetInformer(obj)
+		if err != nil {
+			return err
+		}
+
+		err = c.Watch(&source.Informer{Informer: informer}, &handler.EnqueueRequestForObject{}, p)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func isManaged(meta metav1.Object) bool {
+	// all managed resources are named pod-identity-webhook
+	if meta.GetName() == "pod-identity-webhook" && meta.GetNamespace() == operatorNamespace {
+		return true
+	}
+	// we also care about service-ca rotation
+	if meta.GetName() == "service-ca" && meta.GetNamespace() == "openshift-config-managed" {
```

Comment:
Where is the watch setup for this? I can't spot it.

Reply:
that's because it isn't there.  good catch!

Reply:
https://github.com/openshift/cloud-credential-operator/pull/185#discussion_r414758727